File: backend/src/main/kotlin/com/storepilot/backend/seller/SellerPlan.kt
package com.storepilot.backend.seller

import com.storepilot.backend.common.WireValue
import com.storepilot.backend.common.WireValueEnumConverter
import jakarta.persistence.Converter

/** COD and bank-transfer are Pro-only today (see OrderService.createOrder/StoreService.upsertSettings) — more features are expected to gate on this later. */
enum class SellerPlan(override val wireValue: String) : WireValue {
    FREE("free"),
    PRO("pro"),
}

@Converter(autoApply = true)
class SellerPlanConverter : WireValueEnumConverter<SellerPlan>(SellerPlan.entries.toTypedArray())

File: app/src/types/seller-plan.ts
/** Matches backend SellerPlan — "free" or "pro". Pro unlocks Cash on Delivery + Bank transfer as payment options (more features expected to gate on this later). */
export type SellerPlanTier = "free" | "pro";

/** Matches backend SellerPlanResponse. */
export interface SellerPlanInfo {
  plan: SellerPlanTier;
  currentPeriodEnd?: string;
  cancelAtPeriodEnd: boolean;
  /** Cents — see backend SellerPlan.kt. */
  monthlyPriceCents: number;
  currencyCode: string;
}
